### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by mikebell90**
See https://opentable.atlassian.net/browse/VIS-448

### CODE REVIEW

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
+package com.opentable.metrics.graphite;
+
+import org.slf4j.Logger;
```
Comment (mikebell90) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java]: Copy/paste just for 3 lines... see below. Ugh thhis is hhorribble

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+            if (address.getAddress() == null) {
+                throw new UnknownHostException(address.getHostName());
+            }
+        }
+
+        try {
+            this.socket = socketFactory.createSocket(address.getAddress(), address.getPort());
+        } catch (IOException e) {
+            LOGGER.error("PLEASE REPORT TO ARCHTEAM",e);
```
Comment (mikebell90) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java]: Let's us confirm our suspicion

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java [hunk 3]
```diff
@@ -126,10 +123,8 @@ public GraphiteSender graphiteSender(MetricRegistry metricRegistry) {
             LOG.info("no graphite host; skipping sender initialization");
             return null;
         }
-
-        final Supplier<InetSocketAddress> address = () -> new InetSocketAddress(host, port);
-        final Graphite graphite = new Graphite(address.get());
-        GraphiteSenderWrapper result = new GraphiteSenderWrapper(address, graphite);
+        final Graphite graphite = new Graphite(host, port);
```
Comment (mikebell90) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java]: Switchh to host, port. See Graphite? This forces a new InetAddress each time

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteSenderWrapper.java [hunk 4]
```diff
@@ -57,15 +54,17 @@
...
     private final Counter connectionCloses = new Counter();
-    private final Supplier<InetSocketAddress> address;
+    private final String host;
+    private final int port;
 
     @GuardedBy("this")
     private Graphite delegate; // either connect()ed or null
     @GuardedBy("this")
-    private Instant lastReconnect;
+    private Instant lastReconnect = Instant.now();
```
Comment (mikebell90) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteSenderWrapper.java]: This effectively reenables the one hhour recycle

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteSenderWrapper.java [hunk 5]
```diff
@@ -120,7 +119,7 @@ public synchronized int getFailures() {
     private synchronized Graphite maybeRecycle() throws IOException {
         if (needsReconnectPeriodic()) { //NOPMD
             // Spin up new one
-            Graphite newGraphite = new Graphite(address.get());
```
Comment (SegFaultAX) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteSenderWrapper.java]: This technically didn't need to change. The supplier is going to force re-resolution every time.

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java [hunk 6]
```diff
@@ -126,10 +123,8 @@ public GraphiteSender graphiteSender(MetricRegistry metricRegistry) {
             LOG.info("no graphite host; skipping sender initialization");
             return null;
         }
-
-        final Supplier<InetSocketAddress> address = () -> new InetSocketAddress(host, port);
-        final Graphite graphite = new Graphite(address.get());
```
Comment (SegFaultAX) [inline otj-metrics-core/src/main/java/com/opentable/metrics/graphite/GraphiteConfiguration.java]: I think this line was the source of the bug.

**Comment by mikebell90** [comment]
Most of thhis stuff is pretty meh, so I'm open to discussion

File: otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java (hunk 2 above)
Comment (scottjohnson) [reply otj-metrics-core/src/main/java/com/opentable/metrics/graphite/Graphite.java]: Maybe put in the slack channel name as part of the message?

**Pull request merged**